integration/cli-tests/src/test/java/org/wso2/carbon/esb/cli/CliAPITestCase.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.cli;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.annotations.BeforeClass;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import util.TestUtils;
+
+public class CliAPITestCase {
+
+    protected Log log = LogFactory.getLog(CliAPITestCase.class);
+    BufferedReader reader = null;
+    String line = null;
+    private static final String cliTestApi = "cliTesApi";
+    private static final String cliTestMediatorApi = "cliTestXmltoJsonAPI";
+    File miBuildFilePath;
+
+    {
+        try {
+            TestUtils testUtils = new TestUtils();
+            miBuildFilePath = new File(".." + File.separator + ".." + File.separator + ".." + File.separator
+                    + "cmd" + File.separator + "build" + File.separator + "wso2mi-cli-" + testUtils.getPomVerion()
+                    + File.separator + "bin" + File.separator + "mi");
+        } catch (IOException e) {
+            log.info("Exception = " + e.getMessage());
+        } catch (XmlPullParserException e) {
+            log.info("Exception = " + e.getMessage());
+        }
+    }
+
+    /**
+     * setup the environment to run the tests
+     */
+    @BeforeClass
+    public void setupEnv() throws IOException {
+
+        Process process;
+        String[] setup = { "sh", ".."+ File.separator +"src"+ File.separator +"test"+ File.separator +"java"+ File.separator +"EnvSetup.sh"};
+        process = Runtime.getRuntime().exec(setup);
+
+        try (BufferedReader bufferedreader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+            while ((line = bufferedreader.readLine()) != null) {
+                log.info(line);
+            }
+        }
+    }
+
+    /**
+     * Get information about all the API's
+     */
+    @Test
+    public void miShowAllApiTest() throws IOException {
+        try {
+            ProcessBuilder builder = new ProcessBuilder( miBuildFilePath.getCanonicalPath(), "api" ,"show");
+            Process process = builder.start();
+
+            reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            List<String> lines = new ArrayList();
+            while ((line = reader.readLine()) != null) {
+                lines.add(line);
+            }
```

Comment:
Shall we move this to a method and re use ( from line 80 to 87 or sth ) since it is used in many places